src/main/daml/Daml/Finance/Interface/Claims/Util/Lifecycle.daml

```diff
@@ -74,6 +75,23 @@ lifecycleClaims observableCids acquisitionTime claims events = do
   observe <- collectObservables observableCids
   fmap mconcat . unzip <$> mapA (applyEvents observe acquisitionTime events) claims
 
+-- | Lifecycle a set of claims at specified events.
+lifecycleAndNetClaims : [ContractId Observable.I]          -- ^ The set of observables.
+               -> Time                               -- ^ The input claims' acquisition time.
+               -> [TaggedClaim]                      -- ^ The input claims.
+               -> [Event]                            -- ^ Lifecycle events.
+               -> Update ([TaggedClaim], [Pending])  -- ^ The remaining claims and pending payments.
+lifecycleAndNetClaims observableCids acquisitionTime claims events = do
+  (remaining, pending) <- lifecycleClaims observableCids acquisitionTime claims events
+  let
+    uniqueInstruments = dedup $ map (\c -> c.instrument) pending
+    -- For each unique instrument key:
```

Comment:
You can leverage some of the functions from [DA.List](https://docs.daml.com/daml/stdlib/DA-List.html), e..g. you could do a `groupOn` followed by a `map`. I feel this should be equally efficient but easier to read.

Reply:
Thanks, this grouping also makes it easier to create the other fields of the aggregated Pending nodes.